common/lib/capa/capa/util.py

```diff
@@ -65,24 +65,22 @@ def compare_with_tolerance(student_complex, instructor_complex, tolerance=defaul
         # `tolerance` both equal to infinity. Then, below we would have
         # `inf <= inf` which is a fail. Instead, compare directly.
         return student_complex == instructor_complex
+
+    if isinstance(instructor_complex, complex):
+        if instructor_complex.imag == 0:
+            instructor_complex = instructor_complex.real
+
+    if isinstance(student_complex, float) and isinstance(instructor_complex, float):
```

Comment:
Instead of L69-71, just make one check like this:

```
if isinstance(student_complex, float) and isinstance(instructor_complex, complex) and instructor_complex.imag == 0:
    ...
```